snap/local/start-mysqld-exporter.sh

```diff
@@ -1,6 +1,23 @@
 #!/bin/bash
 
-# For security measures, daemons should not be run as sudo. Execute mysqlrouter as the non-sudo user: snap-daemon.
-exec $SNAP/usr/bin/setpriv --clear-groups --reuid snap_daemon \
-  --regid snap_daemon -- $SNAP/bin/mysqld_exporter --config.my-cnf=$SNAP_DATA/etc/mysql/mysql.cnf "$@"
+USER="$(snapctl get exporter.user)"
```

Comment:
LGTM. Q: Will this work in ROCK/OCI later?

Reply:
It won't.
We have two options:
1. Add more logic to the script, detecting if it's running from a SNAP, and case it's not, retrieve credentials from env's.
2. On the k8s  charm, configure the exporter service pebble layer to call `mysqld_exporter` binary directly, configuring parameters/env there.

tbh, given we are not just setting credentials, but also GUID, UID I prefer option 2.

What's your thoughts @taurus-forever ?

Reply:
2) is OK with me.

JFYI, [MongoDB](https://github.com/canonical/charmed-mongodb-snap/blob/5/edge/snap/local/start-mongodb-exporter.sh#L4). :-)